### DISCUSSION TITLE
[@types/react-datepicker] - add onChange optional type in index.d.ts.

### DISCUSSION THREAD

**Discussion by lokba** [DefinitelyTyped/DefinitelyTyped#62626] [Issues with a @types package] [answered]
I'm using Typescript with React.
When using react-datepicker, I experienced a type error problem when I wanted to make onChange optional.

index.d.ts -  [go to definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-datepicker/index.d.ts)
```typescript
export interface ReactDatePickerProps<
    CustomModifierNames extends string = never,
    WithRange extends boolean | undefined = undefined,
> {
    ...
    onChange(
        date: WithRange extends false | undefined ? Date | null : [Date | null, Date | null],
        event: React.SyntheticEvent<any> | undefined,
    ): void;
    ...
}
```

It seems onChange is not included optional type in index.d.ts.

[a comment by an automated account is not shown]

**Comment by seckie** [comment, marked as answer]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61966#pullrequestreview-1087843865

**Comment by lokba** [comment]
@seckie thanks for the commet!
